graphbolt/include/graphbolt/sampled_subgraph.h

```diff
@@ -0,0 +1,118 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/sampled_subgraph.h
+ * @brief Header file of sampled sub graph.
+ */
+
+#ifndef GRAPHBOLT_SAMPLED_SUBGRAPH_H_
+#define GRAPHBOLT_SAMPLED_SUBGRAPH_H_
+
+#include <torch/custom_class.h>
+#include <torch/torch.h>
+
+namespace graphbolt {
+namespace sampling {
+
+/**
+ * @brief Struct representing a sampled subgraph.
+ *
+ * Example usage:
+ *
+ * Suppose the subgraph has 3 nodes and 9 edges.
+ * ```
+ * auto indptr = torch::tensor({0, 2, 3, 4}, {torch::kInt64});
+ * auto indices = torch::tensor({55, 101, 3, 3}, {torch::kInt64});
+ * auto reverse_row_node_ids = torch::tensor({3, 3, 101}, {torch::kInt64});
+ *
+ * SampledSubgraph sampledSubgraph(indptr, indices, reverse_row_node_ids);
+ * ```
+ *
+ * The `reverse_row_node_ids` indicates that nodes `[3, 3, 101]` in the
+ * original graph are mapped to `[0, 1, 2]` in this subgraph, and because
+ * `reverse_column_node_ids` is `Null`, `{55, 101, 3, 3}` in `indices` is just
+ * the original node ids without compaction.
+ *
+ * If `reverse_column_node_ids = torch::tensor({55, 101, 3}, {torch::kInt64})`,
+ * it would indicate a different mapping for the column nodes. Note this is
+ * inconsistent with row, which is legal, as `3` is mapped to `0` and `1` in the
+ * row while `2` in the column.
+ */
+struct SampledSubgraph : torch::CustomClassHolder {
+ public:
+  /**
+   * @brief Constructor for the SampledSubgraph struct.
+   *
+   * @param indptr CSC format index pointer array.
+   * @param indices CSC format index array.
+   * @param reverse_row_node_ids Row's reverse node ids in the original graph.
+   * @param reverse_column_node_ids Column's reverse node ids in the original
+   * graph.
+   * @param reverse_edge_ids Reverse edge ids in the original graph.
+   * @param type_per_edge Type id of each edge.
+   */
+  SampledSubgraph(
```

Comment:
should the class name be prefixed with `csc` even `indptr` and `indices` are in CSC format? just consistent with `CSCSamplingGraph`?

Reply:
This may be not only used for `CSCSamplingGraph `, so removed 'CSC' from the initial version.